system-helper/flatpak-system-helper.c

```diff
@@ -193,6 +232,160 @@ no_progress_cb (OstreeAsyncProgress *progress, gpointer user_data)
+                                             "org.freedesktop.DBus",
+                                             "GetConnectionCredentials",
+                                             g_variant_new ("(s)", sender),
+                                             NULL, G_DBUS_CALL_FLAGS_NONE, G_MAXINT, NULL, error);
+  if (credentials == NULL)
+    {
+      g_dbus_method_invocation_return_error (invocation, G_DBUS_ERROR, G_DBUS_ERROR_FAILED,
+                                             "Error calling GetConnectionCredentials over D-Bus");
+      return TRUE;
+    }
+
+  g_variant_get (credentials, "(a{sv})", &iter);
+  while (g_variant_iter_loop (iter, "{&sv}", &key, &value))
+    {
+      if (strcmp (key, "UnixUserID") == 0)
+        {
+          uid = g_variant_get_uint32 (value);
+          break;
+        }
+    }
+
+  return uid;
+}
+
+static void
+mounts_changed (GUnixMountMonitor *monitor, gpointer user_data)
+{
+  FlatpakSystemHelperOngoingPull *pull = (FlatpakSystemHelperOngoingPull *) user_data;
+  g_autoptr(GList) mounts = NULL;
+  GList *l = NULL;
+
+  mounts = g_unix_mounts_get (&pull->current_timestamp);
+  for (l = g_list_reverse (mounts); l != NULL; l = l->next)
+    {
+      if (g_strcmp0 (pull->user_folder_name, g_unix_mount_get_mount_path (l->data)) == 0)
+        {
+          flatpak_system_helper_complete_create_pull_cache_dir (pull->object,
+                                                                pull->invocation,
+                                                                pull->user_folder_name);
+	  break;
+        }
+    }
+  // TODO: Way to return error if mount isn't successful or timeout
+}
+
+static gboolean
+handle_create_pull_cache_dir (FlatpakSystemHelper   *object,
+                              GDBusMethodInvocation *invocation,
+                              const gchar           *arg_ref,
+                              const gchar           *optional_commit,
+                              const gchar           *arg_installation)
+{
+  GDBusConnection *connection = g_dbus_method_invocation_get_connection (invocation);
+  const gchar *sender = g_dbus_method_invocation_get_sender (invocation);
+  FlatpakSystemHelperOngoingPull *new_pull;
+  g_autoptr(FlatpakDir) system = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autofree gchar *user_folder = NULL;
+  g_autofree gchar *root_folder = NULL;
+  g_autofree gchar *flatpak_dir = NULL;
+  g_autofree gchar *bindfs_path = NULL;
+  guint uid_pull_count = 0;
+  uid_t uid;
+
+  system = dir_get_system (arg_installation, &error);
+  if (system == NULL)
+    {
+      g_dbus_method_invocation_return_gerror (invocation, error);
+      return TRUE;
+    }
+
+  if (ongoing_pulls == NULL)
+    ongoing_pulls = g_ptr_array_new ();
+
+  for (guint i = 0; i < ongoing_pulls->len; i++)
+    {
+      FlatpakSystemHelperOngoingPull *pull_tmp;
+      pull_tmp = g_ptr_array_index (ongoing_pulls, i);
+      if (pull_tmp->uid == uid)
```

Comment:
`uid` isn't defined until after this loop

Reply:
This is one of those moments: "What was I thinking" !